libraries/chain/webassembly/runtimes/host_function_registrator.cpp

```diff
@@ -0,0 +1,342 @@
+      return {BOOST_HANA_STRING("env"), BOOST_HANA_STRING(#NAME)};                                                     \
+   }                                                                                                                   \
+   inline static auto NAME##_registrator = NAME##_registrator_impl();
+
+#define REGISTER_CF_HOST_FUNCTION(NAME, ...)                                                                           \
+   static host_function_registrator<&interface::NAME, core_precondition, ##__VA_ARGS__> NAME##_registrator_impl() {    \
+      return {BOOST_HANA_STRING("env"), BOOST_HANA_STRING(#NAME)};                                                     \
+   }                                                                                                                   \
+   inline static auto NAME##_registrator = NAME##_registrator_impl();
+
+#define REGISTER_LEGACY_HOST_FUNCTION(NAME, ...)                                                                       \
+   static host_function_registrator<&interface::NAME, legacy_static_check_wl_args, context_aware_check, ##__VA_ARGS__> \
+       NAME##_registrator_impl() {                                                                                     \
+      return {BOOST_HANA_STRING("env"), BOOST_HANA_STRING(#NAME)};                                                     \
+   }                                                                                                                   \
+   inline static auto NAME##_registrator = NAME##_registrator_impl();
+
+#define REGISTER_LEGACY_CF_HOST_FUNCTION(NAME, ...)                                                                    \
+   static host_function_registrator<&interface::NAME, legacy_static_check_wl_args, ##__VA_ARGS__>                      \
+       NAME##_registrator_impl() {                                                                                     \
+      return {BOOST_HANA_STRING("env"), BOOST_HANA_STRING(#NAME)};                                                     \
+   }                                                                                                                   \
+   inline static auto NAME##_registrator = NAME##_registrator_impl();
+
+#define REGISTER_LEGACY_CF_ONLY_HOST_FUNCTION(NAME, ...)                                                               \
+   static host_function_registrator<&interface::NAME, legacy_static_check_wl_args, context_free_check, ##__VA_ARGS__>  \
+       NAME##_registrator_impl() {                                                                                     \
+      return {BOOST_HANA_STRING("env"), BOOST_HANA_STRING(#NAME)};                                                     \
+   }                                                                                                                   \
+   inline static auto NAME##_registrator = NAME##_registrator_impl();
+
+REGISTER_LEGACY_CF_ONLY_HOST_FUNCTION(get_context_free_data)
```

Comment:
Please add the comment grouping that is the same as in `interface.hpp` to help navigate this better.